public/app/plugins/panel/datagrid/DataGridPanel.tsx

```diff
@@ -108,7 +108,11 @@ export function DataGridPanel({ options, data, id, fieldConfig, width, height }:
 
   const addNewRow = () => {
     const newFrame = new MutableDataFrame(frame);
-    newFrame.appendRow(new Array(newFrame.fields.length).fill(null));
+
+    newFrame.fields.map((field) => {
+      field.values = new ArrayVector([...field.values.toArray(), null]);
+    });
```

Comment:
The field values have been modified recently and the received field values aren't ArrayVectors anymore, but ArrayPropertyVector's. I'm not sure why it has changed but the entire datagrid is built around arrayVectors and changing the functionality now is not feasible.

Reply:
Maybe add a TODO message to follow up on this.